### PR TITLE
Refactor earlybirds page

### PR DESCRIPTION
**Pull request by mingsorawit**
### Problems
18/8
- create ErrorStateMessage component 
- change input fields placeholder color
- disable input fields and button while submitting
- use async await in onSubmit
- create renderParagraph function
- change to flex(wrapper) and remove top,left,transform(main_container)
- remove height for main_container(desktop and every responsive version)
//
- create and use Text and Spacer component
- remove unused class

19/8
- lighter font weight for error message
- use gap and spacer instead of margin

20/8
- move arrow in country select field to match the design (use background image)
- validate phone number using regex
- change some error message

21/8
- change index page animation sequence
- add more padding for portrait small phone screen (except first paragraph)

22/8 
- make fail message in center
- edit spacer between fail message and remarks to 80px for every device

### Preflight checklist

- [ ] **Manual testing** has been performed according to the acceptance criteria.
- [ ] _(heavy style change)_ The solution has been tested on the major browser (Chrome, Firefox, Safari, Microsoft Edge).

### CODE REVIEW

File: styles/globals.css [hunk 1]
```diff
@@ -57,3 +57,8 @@ input {
   font-size: 16px;
   font-family: 'Lexend Deca';
 }
+
+button {
+  font-family: 'Lexend Deca';
+  font-size: 16px;
+}
```
Comment (oukunanan) [inline styles/globals.css:64]:
```suggestion
input, button {
  font-family: 'Lexend Deca';
  font-size: 16px;
}
```

File: components/early_birds/ErrorStateMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import styles from './ErrorStateMessage.module.css'
+import Text from '../Text'
+type Props = {
+  name: string
+}
+
+export default function ErrorStateMessage(props: Props) {
+  return (
+    <div className={styles.error_message}>
+      <Text color="#f65129" family="Assistant">
```
Comment (oukunanan) [inline components/early_birds/ErrorStateMessage.tsx:11]: Seems the design error font-weight is lighter than yours. `Assistant` font has a lot of weight [here](https://fonts.google.com/specimen/Assistant), please take a look kub.

**Currently**
Screen Shot 2564-08-19 at 10 11 48

**Design**
Screen Shot 2564-08-19 at 10 12 27

File: components/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  color?: string
+  family?: 'Lexend Deca' | 'Assistant' | 'Prompt'
+  children: React.ReactNode
+}
+
+export default function Text(props: Props) {
+  return (
+    <span
+      style={{
+        fontSize: `${props.size}px`,
```
Comment (oukunanan) [inline components/Text.tsx:14]: Can we remove `undefinedpx` ?
Screen Shot 2564-08-19 at 10 19 26

File: components/early_birds/SignupForm.tsx [hunk 4]
```diff
@@ -61,15 +57,12 @@ export default function SignupForm() {
                   props.touched.fullName &&
                   styles.error_input_field
               )}
+              disabled={props.isSubmitting}
             />
-            <div className={styles.error_message}>
-              <ErrorMessage name="fullName" />
-            </div>
+            <ErrorStateMessage name="fullName" />
```
Comment (oukunanan) [inline components/early_birds/SignupForm.tsx:62]: Do not render if an error not occur yet.
Screen Shot 2564-08-19 at 10 23 48

File: components/early_birds/ErrorStateMessage.module.css [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.error_message {
+  height: 20px;
+  width: 100%;
+  margin-top: 5px;
+  margin-bottom: 12px;
```
Comment (oukunanan) [inline components/early_birds/ErrorStateMessage.module.css:5]: Please NOT add `margin` to the high-level components. Maybe we can use  [gap](https://coryrylan.com/blog/css-gap-space-with-flexbox) for creating the spacing.

File: pages/earlybirds.tsx [hunk 6]
```diff
@@ -70,65 +120,30 @@ export default function Earlybird() {
...
-          </Link>
-          . We reserve the right to refuse providing offers and/or promotions at
-          any time, either for particular individuals or organizations at our
-          discretion.
+          <Text size={12} color="#535353">
+            Remarks: This campaign is applicable for stores with online sales
+            value of over THB 20k/month. By registering this campaign, you agree
+            to Beam&apos;s{' '}
+            <Link href="https://beamdata.co/privacy/" passHref>
+              <a>Privacy Policy</a>
```
Comment (oukunanan) [inline pages/earlybirds.tsx:137]: Do we really need anchor tag `<a />` ?

**Comment by oukunanan** [comment]
Do not forget to fixed this one kub
Screen Shot 2564-08-19 at 10 36 13

File: components/Text.tsx (hunk 3 above)
Comment (mingsorawit) [reply components/Text.tsx]: use 16px instead when props.size is undefined

File: components/early_birds/SignupForm.tsx (hunk 4 above)
Comment (mingsorawit) [reply components/early_birds/SignupForm.tsx]: changed to spacer

**Pull request merged**